exercises/sam_jenkins/ex01_basics/src/main.cpp

```diff
@@ -0,0 +1,44 @@
+#include <iostream>
+#include <fstream>
+#include <unordered_map>
+#include <iomanip>
+#include <vector>
+#include "mapFunctions.h"
+
+
+using std::ifstream; using std::string; using std::cout; using std::endl; using std::istreambuf_iterator; 
+using std::unordered_map; using std::vector; using std::ofstream;
+
+
+int main(int argc, char **argv) {
+	//check for valid number of arguements
+	if (argc !=2) {
+		cout << "This Program requires a single input" << endl;
+			return 1;
+	}
+	//open the input stream
+	ifstream in(argv[1]);
+	if (in) {
```

Comment:
Instead of having most of your code in an if statement, use an [early exit](https://llvm.org/docs/CodingStandards.html#use-early-exits-and-continue-to-simplify-code) instead:

```cpp
if (!in) {
    cout << "could not find file: " << argv[1] << endl;
    return 1;
}

...